### PR TITLE
ci: Full deployment automation

### PR DESCRIPTION
**Pull request by VladislavFitz**
**Summary**

- Automatic changelog and release note generation using _polyglot_ tool
- Add comments to **generate_version** script

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -48,37 +48,59 @@ end
...
+  	api_token: ENV["GITHUB_API_TOKEN"],
+  	name: "#{new_version}",
+  	tag_name: "#{new_version}",
+  	description: (File.read("../release_notes.md") rescue "No release notes provided"),
+  	is_draft: true,
+  	commitish: "master",
+  	upload_assets: []
+  )
+
+  # Remove deployment trigger tag if needed
```
Comment (spinach) [inline fastlane/Fastfile:99]: Love this!! Automation for the win :)

File: fastlane/Fastfile [hunk 2]
```diff
@@ -48,37 +48,59 @@ end
...
+  sh("cd ..; ./generate_version #{new_version}")
 
   branchName = "version-#{new_build_number}"
   sh("git checkout -b #{branchName}")
 
   git_commit(
-    path: ["InstantSearch.podspec", "./Sources/InstantSearchCore/Helper/Version+Current.swift"],
-    message: "chore: Version #{new_build_number}"
+    path: ["InstantSearch.podspec", "./Sources/InstantSearchCore/Helper/Version+Current.swift", "CHANGELOG.md"],
+    message: "chore: update version to #{new_version} [skip ci]"
```
Comment (spinach) [inline fastlane/Fastfile:71]: good one for the skip ci 👍

File: fastlane/Fastfile [hunk 3]
```diff
@@ -48,37 +48,59 @@ end
...
   )
 
+  set_github_release(
+  	repository_name: "algolia/instantsearch-ios",
+  	api_token: ENV["GITHUB_API_TOKEN"],
+  	name: "#{new_version}",
+  	tag_name: "#{new_version}",
+  	description: (File.read("../release_notes.md") rescue "No release notes provided"),
+  	is_draft: true,
+  	commitish: "master",
```
Comment (spinach) [inline fastlane/Fastfile:95]: nitpicky: we use `"master"` 3 times in the script, we can perhaps extract in variable `target_branch` or something

**Review by spinach — APPROVED** [review]

File: fastlane/Fastfile (hunk 3 above)
Comment (VladislavFitz) [reply fastlane/Fastfile]: Fixed ✔️

**Pull request merged**